webapp/utils/bitcoin.ts

```diff
@@ -23,3 +23,16 @@ export const calculateDepositOutputIndex = (
   transactionReceipt.vout.findIndex(
     equalsBitcoinCustodyAddress(bitcoinCustodyAddress),
   )
+
+/**
+ * The Bitcoin API we're using (esplora-client) sometimes returns data with +- 1 hour of range.
+ * This means that under certain conditions, the timestamp may be visible to the user as "in X minutes",
+ * meaning the date looks like it is in the future. This function caps the timestamp to "now".
+ * If the user ever resyncs, and the timestamp appeared in the past, then we save that.
+ * See https://github.com/hemilabs/ui-monorepo/issues/692
+ */
+export const getBitcoinTimestamp = function (timestamp: bigint | number) {
+  // timestamps from btc are saved in unix format
+  const now = Math.floor(new Date().getTime() / 1000)
+  return Math.min(now, Number(timestamp))
+}
```

Comment:
Does this function receive either bigints or numbers? We may want to stick to just one input type, the one in `*.status.blockTime`.

Reply:
Agree, the type is `number`, so `bigint` was dropped here [76bb03e](https://github.com/hemilabs/ui-monorepo/pull/718/commits/76bb03e49bfa923b401fbad5dbf470cd535948fd)